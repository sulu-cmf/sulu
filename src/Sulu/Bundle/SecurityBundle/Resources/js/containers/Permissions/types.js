// @flow
export type ContextPermission = {
    context: string,
    id: ?number,
    permissions: {[string]: boolean},
};
