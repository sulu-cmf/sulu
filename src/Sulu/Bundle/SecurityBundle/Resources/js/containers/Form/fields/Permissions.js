// @flow
import React from 'react';
import {computed} from 'mobx';
import {observer} from 'mobx-react';
import type {FieldTypeProps} from 'sulu-admin-bundle/types';
import PermissionsContainer from '../../Permissions';
import type {ContextPermission} from '../../Permissions';

type Props = FieldTypeProps<?Array<ContextPermission>>;

@observer
export default class Permissions extends React.Component<Props> {
    @computed get system(): ?string {
        const {formInspector} = this.props;
        const system = formInspector.getValueByPath('/system');

        if (!system || typeof system !== 'string') {
            return null;
        }

        return system;
    }

    handleChange = (value: Array<ContextPermission>) => {
        const {onChange, onFinish} = this.props;
        onChange(value);
        onFinish();
    };

    render() {
        const {value} = this.props;

        if (!this.system) {
            return null;
        }

        return (
            <PermissionsContainer onChange={this.handleChange} system={this.system} value={value ? value : []} />
        );
    }
}
