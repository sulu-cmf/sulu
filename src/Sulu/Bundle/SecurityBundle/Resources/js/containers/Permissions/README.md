The `Permissions` container is used to display the available context permissions in the given system.
