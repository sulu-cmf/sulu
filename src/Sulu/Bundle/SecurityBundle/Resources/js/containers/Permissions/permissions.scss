.matrix-container {
    margin-bottom: 40px;

    h3 {
        margin-top: 30px;
    }
}

.select-container {
    margin-bottom: 20px;
    width: 50%;
}
