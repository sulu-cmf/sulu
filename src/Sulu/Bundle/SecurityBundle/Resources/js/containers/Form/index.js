// @flow
import Permissions from './fields/Permissions';
import RoleAssignments from './fields/RoleAssignments';

export {
    Permissions,
    RoleAssignments,
};
