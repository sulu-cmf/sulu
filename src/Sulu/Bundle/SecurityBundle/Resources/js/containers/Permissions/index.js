// @flow
import Permissions from './Permissions';
import type {ContextPermission} from './types';

export type {ContextPermission};
export default Permissions;
