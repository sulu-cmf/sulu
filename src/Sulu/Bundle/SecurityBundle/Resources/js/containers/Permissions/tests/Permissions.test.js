// @flow
import React from 'react';
import {mount} from 'enzyme';
import ResourceListStore from 'sulu-admin-bundle/stores/ResourceListStore';
import {MultiSelect} from 'sulu-admin-bundle/containers';
import {userStore} from 'sulu-admin-bundle/stores';
import Permissions from '../Permissions';
import type {ContextPermission} from '../types';
import type {SecurityContextGroups} from '../../../stores/SecurityContextStore/types';
import securityContextStore from '../../../stores/SecurityContextStore/SecurityContextStore';
import PermissionMatrix from '../PermissionMatrix';

jest.mock('sulu-admin-bundle/stores/ResourceListStore', () => jest.fn());

jest.mock('../../../stores/SecurityContextStore/SecurityContextStore', () => ({
    loadSecurityContextGroups: jest.fn(() => Promise.resolve()),
}));

jest.mock('sulu-admin-bundle/stores', () => ({
    userStore: {
        user: {
            locale: 'en',
        },
    },
}));

jest.mock('sulu-admin-bundle/utils', () => ({
    translate: (key) => key,
}));

jest.mock('sulu-admin-bundle/utils/Translator', () => ({
    translate: (key) => key,
}));

test('Render with minimal', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        expect(securityContextStore.loadSecurityContextGroups).toBeCalledWith('Sulu');
        permissions.update();
        expect(permissions.render()).toMatchSnapshot();
    });
});

test('Should trigger onChange correctly', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    const onChange = jest.fn();
    const permissions = mount(
        <Permissions
            onChange={onChange}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        permissions.update();
        const newContextPermissions: Array<ContextPermission> = [
            {
                id: 1,
                context: 'sulu.contact.people',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 2,
                context: 'sulu.contact.organizations',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': false,
                },
            },
        ];
        permissions.find(PermissionMatrix).at(0).instance().props.onChange(newContextPermissions);
        expect(onChange).toBeCalledWith(newContextPermissions);
    });
});

test('Render with empty webspace section', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view', 'add', 'edit', 'delete', 'live', 'security'],
            'sulu.webspaces.#webspace#.analytics': ['view', 'add', 'edit', 'delete'],
            'sulu.webspaces.#webspace#.default-snippets': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    // $FlowFixMe
    ResourceListStore.mockImplementation(function() {
        this.loading = false;
        this.data = [
            {
                'key': 'example',
                'name': 'Example',
            },
            {
                'key': 'example2',
                'name': 'Example 2',
            },
            {
                'key': 'example3',
                'name': 'Example 3!',
            },
        ];
    });

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        expect(securityContextStore.loadSecurityContextGroups).toBeCalledWith('Sulu');
        permissions.update();

        // Currently we have to load each child separately, because of a bug in enzyme.
        // TODO: https://github.com/airbnb/enzyme/issues/1213
        const permissionChildren = permissions.children();
        expect(permissionChildren.at(0).render()).toMatchSnapshot();
        expect(permissionChildren.at(1).render()).toMatchSnapshot();
        expect(permissionChildren.at(2).render()).toMatchSnapshot();
        expect(permissionChildren.at(3).render()).toMatchSnapshot();
    });
});

test('Render with webspace section', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 3,
            context: 'sulu.webspaces.example',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
                'live': false,
                'security': false,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view', 'add', 'edit', 'delete', 'live', 'security'],
            'sulu.webspaces.#webspace#.analytics': ['view', 'add', 'edit', 'delete'],
            'sulu.webspaces.#webspace#.default-snippets': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    // $FlowFixMe
    ResourceListStore.mockImplementation(function() {
        this.loading = false;
        this.data = [
            {
                'key': 'example',
                'name': 'Example',
            },
            {
                'key': 'example2',
                'name': 'Example 2',
            },
            {
                'key': 'example3',
                'name': 'Example 3!',
            },
        ];
    });

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        expect(securityContextStore.loadSecurityContextGroups).toBeCalledWith('Sulu');
        permissions.update();

        // Currently we have to load each child separately, because of a bug in enzyme.
        // TODO: https://github.com/airbnb/enzyme/issues/1213
        const permissionChildren = permissions.children();
        expect(permissionChildren.at(0).render()).toMatchSnapshot();
        expect(permissionChildren.at(1).render()).toMatchSnapshot();
        expect(permissionChildren.at(2).render()).toMatchSnapshot();
        expect(permissionChildren.at(3).render()).toMatchSnapshot();
    });
});

test('Should trigger onChange correctly when changing something in the webspace section', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 3,
            context: 'sulu.webspaces.example',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
                'live': false,
                'security': false,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view', 'add', 'edit', 'delete', 'live', 'security'],
            'sulu.webspaces.#webspace#.analytics': ['view', 'add', 'edit', 'delete'],
            'sulu.webspaces.#webspace#.default-snippets': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    // $FlowFixMe
    ResourceListStore.mockImplementation(function() {
        this.loading = false;
        this.data = [
            {
                'key': 'example',
                'name': 'Example',
            },
            {
                'key': 'example2',
                'name': 'Example 2',
            },
            {
                'key': 'example3',
                'name': 'Example 3!',
            },
        ];
    });

    const onChange = jest.fn();
    const permissions = mount(
        <Permissions
            onChange={onChange}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        permissions.update();
        const newContextPermissions: Array<ContextPermission> = [
            {
                id: 1,
                context: 'sulu.contact.people',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 2,
                context: 'sulu.contact.organizations',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 3,
                context: 'sulu.webspaces.example',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': false,
                    'edit': true,
                    'live': false,
                    'security': false,
                },
            },
        ];
        permissions.find(PermissionMatrix).at(0).instance().props.onChange(newContextPermissions);
        expect(onChange).toBeCalledWith(newContextPermissions);
    });
});

test('Should trigger onChange correctly when a webspace is added', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 3,
            context: 'sulu.webspaces.example',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
                'live': false,
                'security': false,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view', 'add', 'edit', 'delete', 'live', 'security'],
            'sulu.webspaces.#webspace#.analytics': ['view', 'add', 'edit', 'delete'],
            'sulu.webspaces.#webspace#.default-snippets': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    // $FlowFixMe
    ResourceListStore.mockImplementation(function() {
        this.loading = false;
        this.data = [
            {
                'key': 'example',
                'name': 'Example',
            },
            {
                'key': 'example2',
                'name': 'Example 2',
            },
            {
                'key': 'example3',
                'name': 'Example 3!',
            },
        ];
    });

    const onChange = jest.fn();
    const permissions = mount(
        <Permissions
            onChange={onChange}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        permissions.update();
        permissions.find(MultiSelect).at(0).instance().props.onChange(['example', 'example3']);

        const expectedNewValue: Array<ContextPermission> = [
            {
                id: 1,
                context: 'sulu.contact.people',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 2,
                context: 'sulu.contact.organizations',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 3,
                context: 'sulu.webspaces.example',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                    'live': false,
                    'security': false,
                },
            },
            {
                id: undefined,
                context: 'sulu.webspaces.example3',
                permissions: {
                    'view': false,
                    'delete': false,
                    'add': false,
                    'edit': false,
                    'live': false,
                    'security': false,
                },
            },
            {
                id: undefined,
                context: 'sulu.webspaces.example3.analytics',
                permissions: {
                    'view': false,
                    'delete': false,
                    'add': false,
                    'edit': false,
                },
            },
            {
                id: undefined,
                context: 'sulu.webspaces.example3.default-snippets',
                permissions: {
                    'view': false,
                    'delete': false,
                    'add': false,
                    'edit': false,
                },
            },
        ];

        expect(onChange).toBeCalledWith(expectedNewValue);
    });
});

test('Should trigger onChange correctly when a webspace is removed', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 3,
            context: 'sulu.webspaces.example',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
                'live': false,
                'security': false,
            },
        },
        {
            id: 4,
            context: 'sulu.webspaces.example3',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
                'live': false,
                'security': true,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view', 'add', 'edit', 'delete', 'live', 'security'],
            'sulu.webspaces.#webspace#.analytics': ['view', 'add', 'edit', 'delete'],
            'sulu.webspaces.#webspace#.default-snippets': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    // $FlowFixMe
    ResourceListStore.mockImplementation(function() {
        this.loading = false;
        this.data = [
            {
                'key': 'example',
                'name': 'Example',
            },
            {
                'key': 'example2',
                'name': 'Example 2',
            },
            {
                'key': 'example3',
                'name': 'Example 3!',
            },
        ];
    });

    const onChange = jest.fn();
    const permissions = mount(
        <Permissions
            onChange={onChange}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        permissions.update();
        permissions.find(MultiSelect).at(0).instance().props.onChange(['example3']);

        const expectedNewValue: Array<ContextPermission> = [
            {
                id: 1,
                context: 'sulu.contact.people',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 2,
                context: 'sulu.contact.organizations',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                },
            },
            {
                id: 4,
                context: 'sulu.webspaces.example3',
                permissions: {
                    'view': true,
                    'delete': true,
                    'add': true,
                    'edit': true,
                    'live': false,
                    'security': true,
                },
            },
        ];

        expect(onChange).toBeCalledWith(expectedNewValue);
    });
});

test('Should trigger a mobx autorun if the prop system changes', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={value}
        />
    );

    return promise.then(() => {
        // update with the same system, nothing should happen
        // update it with a other system it should trigger a reload
        permissions.setProps({system: 'Sulu'});
        permissions.setProps({system: 'Other-System'});

        expect(securityContextStore.loadSecurityContextGroups).toHaveBeenCalledWith('Sulu');
        expect(securityContextStore.loadSecurityContextGroups).toHaveBeenCalledWith('Other-System');
        expect(securityContextStore.loadSecurityContextGroups).toHaveBeenCalledTimes(2);
    });
});

test('Pass correct apiOptions to MultiSelect', () => {
    const securityContextGroups: SecurityContextGroups = {
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={[]}
        />
    );

    return promise.then(() => {
        permissions.update();
        expect(permissions.find(MultiSelect).prop('apiOptions')).toEqual({
            checkForPermissions: 0,
            locale: 'en',
        });
    });
});

test('Pass correct locale to MultiSelect', () => {
    const securityContextGroups: SecurityContextGroups = {
        'Webspaces': {
            'sulu.webspaces.#webspace#': ['view'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    userStore.user = {
        id: 1,
        locale: 'de',
        settings: [],
        username: 'Test',
    };

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={[]}
        />
    );

    return promise.then(() => {
        permissions.update();
        expect(permissions.find(MultiSelect).prop('apiOptions')).toEqual({
            checkForPermissions: 0,
            locale: 'de',
        });
    });
});

test('Dispose autorun on unmount', () => {
    const value: Array<ContextPermission> = [
        {
            id: 1,
            context: 'sulu.contact.people',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
        {
            id: 2,
            context: 'sulu.contact.organizations',
            permissions: {
                'view': true,
                'delete': true,
                'add': true,
                'edit': true,
            },
        },
    ];

    const securityContextGroups: SecurityContextGroups = {
        'Contacts': {
            'sulu.contact.people': ['view', 'add', 'edit', 'delete'],
            'sulu.contact.organizations': ['view', 'add', 'edit', 'delete'],
        },
    };
    const promise = Promise.resolve(securityContextGroups);
    securityContextStore.loadSecurityContextGroups.mockReturnValue(promise);

    const permissions = mount(
        <Permissions
            onChange={jest.fn()}
            system="Sulu"
            value={value}
        />
    );

    permissions.update();

    const systemDisposerSpy = jest.fn();
    permissions.instance().systemDisposer = systemDisposerSpy;
    permissions.unmount();

    expect(systemDisposerSpy).toBeCalledWith();
});
