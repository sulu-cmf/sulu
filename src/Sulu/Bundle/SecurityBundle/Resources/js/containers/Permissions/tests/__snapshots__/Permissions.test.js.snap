// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render with empty webspace section 1`] = `
<h2>
  Webspaces
</h2>
`;

exports[`Render with empty webspace section 2`] = `
<div
  class="selectContainer"
>
  <div
    class="select"
  >
    <button
      class="displayValue default"
      type="button"
    >
      <div
        aria-label="sulu_admin.none_selected"
        class="croppedText"
        title="sulu_admin.none_selected"
      >
        <div
          aria-hidden="true"
          class="front"
        >
          sulu_admin.n
        </div>
        <div
          aria-hidden="true"
          class="back"
        >
          <span>
            one_selected
          </span>
        </div>
        <div
          class="whole"
        >
          sulu_admin.none_selected
        </div>
      </div>
      <span
        aria-label="su-angle-down"
        class="toggle su-angle-down"
      />
    </button>
  </div>
</div>
`;

exports[`Render with empty webspace section 3`] = `
<div
  class="matrixContainer"
/>
`;

exports[`Render with empty webspace section 4`] = `
<div
  class="matrixContainer"
>
  <h2>
    Contacts
  </h2>
  <div
    class="matrix"
  >
    <div
      class="row"
    >
      <div>
        people
      </div>
      <div>
        <div
          class="item selected"
          title="sulu_security.view"
        >
          <span
            aria-label="su-eye"
            class="su-eye"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.add"
        >
          <span
            aria-label="su-plus-circle"
            class="su-plus-circle"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.edit"
        >
          <span
            aria-label="su-pen"
            class="su-pen"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.delete"
        >
          <span
            aria-label="su-trash-alt"
            class="su-trash-alt"
          />
        </div>
        <button
          class="rowButton"
        >
          sulu_admin.deactivate_all
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div>
        organizations
      </div>
      <div>
        <div
          class="item selected"
          title="sulu_security.view"
        >
          <span
            aria-label="su-eye"
            class="su-eye"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.add"
        >
          <span
            aria-label="su-plus-circle"
            class="su-plus-circle"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.edit"
        >
          <span
            aria-label="su-pen"
            class="su-pen"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.delete"
        >
          <span
            aria-label="su-trash-alt"
            class="su-trash-alt"
          />
        </div>
        <button
          class="rowButton"
        >
          sulu_admin.deactivate_all
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render with minimal 1`] = `
<div
  class="matrixContainer"
>
  <h2>
    Contacts
  </h2>
  <div
    class="matrix"
  >
    <div
      class="row"
    >
      <div>
        people
      </div>
      <div>
        <div
          class="item selected"
          title="sulu_security.view"
        >
          <span
            aria-label="su-eye"
            class="su-eye"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.add"
        >
          <span
            aria-label="su-plus-circle"
            class="su-plus-circle"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.edit"
        >
          <span
            aria-label="su-pen"
            class="su-pen"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.delete"
        >
          <span
            aria-label="su-trash-alt"
            class="su-trash-alt"
          />
        </div>
        <button
          class="rowButton"
        >
          sulu_admin.deactivate_all
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div>
        organizations
      </div>
      <div>
        <div
          class="item selected"
          title="sulu_security.view"
        >
          <span
            aria-label="su-eye"
            class="su-eye"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.add"
        >
          <span
            aria-label="su-plus-circle"
            class="su-plus-circle"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.edit"
        >
          <span
            aria-label="su-pen"
            class="su-pen"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.delete"
        >
          <span
            aria-label="su-trash-alt"
            class="su-trash-alt"
          />
        </div>
        <button
          class="rowButton"
        >
          sulu_admin.deactivate_all
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render with webspace section 1`] = `
<h2>
  Webspaces
</h2>
`;

exports[`Render with webspace section 2`] = `
<div
  class="selectContainer"
>
  <div
    class="select"
  >
    <button
      class="displayValue default"
      type="button"
    >
      <div
        aria-label="Example"
        class="croppedText"
        title="Example"
      >
        <div
          aria-hidden="true"
          class="front"
        >
          Exam
        </div>
        <div
          aria-hidden="true"
          class="back"
        >
          <span>
            ple
          </span>
        </div>
        <div
          class="whole"
        >
          Example
        </div>
      </div>
      <span
        aria-label="su-angle-down"
        class="toggle su-angle-down"
      />
    </button>
  </div>
</div>
`;

exports[`Render with webspace section 3`] = `
<div
  class="matrixContainer"
>
  <div
    class="matrixContainer"
  >
    <h3>
      example
    </h3>
    <div
      class="matrix"
    >
      <div
        class="row"
      >
        <div>
          example
        </div>
        <div>
          <div
            class="item selected"
            title="sulu_security.view"
          >
            <span
              aria-label="su-eye"
              class="su-eye"
            />
          </div>
          <div
            class="item selected"
            title="sulu_security.add"
          >
            <span
              aria-label="su-plus-circle"
              class="su-plus-circle"
            />
          </div>
          <div
            class="item selected"
            title="sulu_security.edit"
          >
            <span
              aria-label="su-pen"
              class="su-pen"
            />
          </div>
          <div
            class="item selected"
            title="sulu_security.delete"
          >
            <span
              aria-label="su-trash-alt"
              class="su-trash-alt"
            />
          </div>
          <div
            class="item"
            title="sulu_security.live"
          >
            <span
              aria-label="fa-signal"
              class="fa fa-signal"
            />
          </div>
          <div
            class="item"
            title="sulu_security.security"
          >
            <span
              aria-label="su-lock"
              class="su-lock"
            />
          </div>
          <button
            class="rowButton"
          >
            sulu_admin.deactivate_all
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div>
          example.analytics
        </div>
        <div>
          <div
            class="item"
            title="sulu_security.view"
          >
            <span
              aria-label="su-eye"
              class="su-eye"
            />
          </div>
          <div
            class="item"
            title="sulu_security.add"
          >
            <span
              aria-label="su-plus-circle"
              class="su-plus-circle"
            />
          </div>
          <div
            class="item"
            title="sulu_security.edit"
          >
            <span
              aria-label="su-pen"
              class="su-pen"
            />
          </div>
          <div
            class="item"
            title="sulu_security.delete"
          >
            <span
              aria-label="su-trash-alt"
              class="su-trash-alt"
            />
          </div>
          <button
            class="rowButton"
          >
            sulu_admin.activate_all
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div>
          example.default-snippets
        </div>
        <div>
          <div
            class="item"
            title="sulu_security.view"
          >
            <span
              aria-label="su-eye"
              class="su-eye"
            />
          </div>
          <div
            class="item"
            title="sulu_security.add"
          >
            <span
              aria-label="su-plus-circle"
              class="su-plus-circle"
            />
          </div>
          <div
            class="item"
            title="sulu_security.edit"
          >
            <span
              aria-label="su-pen"
              class="su-pen"
            />
          </div>
          <div
            class="item"
            title="sulu_security.delete"
          >
            <span
              aria-label="su-trash-alt"
              class="su-trash-alt"
            />
          </div>
          <button
            class="rowButton"
          >
            sulu_admin.activate_all
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render with webspace section 4`] = `
<div
  class="matrixContainer"
>
  <h2>
    Contacts
  </h2>
  <div
    class="matrix"
  >
    <div
      class="row"
    >
      <div>
        people
      </div>
      <div>
        <div
          class="item selected"
          title="sulu_security.view"
        >
          <span
            aria-label="su-eye"
            class="su-eye"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.add"
        >
          <span
            aria-label="su-plus-circle"
            class="su-plus-circle"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.edit"
        >
          <span
            aria-label="su-pen"
            class="su-pen"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.delete"
        >
          <span
            aria-label="su-trash-alt"
            class="su-trash-alt"
          />
        </div>
        <button
          class="rowButton"
        >
          sulu_admin.deactivate_all
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div>
        organizations
      </div>
      <div>
        <div
          class="item selected"
          title="sulu_security.view"
        >
          <span
            aria-label="su-eye"
            class="su-eye"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.add"
        >
          <span
            aria-label="su-plus-circle"
            class="su-plus-circle"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.edit"
        >
          <span
            aria-label="su-pen"
            class="su-pen"
          />
        </div>
        <div
          class="item selected"
          title="sulu_security.delete"
        >
          <span
            aria-label="su-trash-alt"
            class="su-trash-alt"
          />
        </div>
        <button
          class="rowButton"
        >
          sulu_admin.deactivate_all
        </button>
      </div>
    </div>
  </div>
</div>
`;
